internal/validate/helpers.go

```diff
@@ -79,3 +90,85 @@ func ReadFile(ctx context.Context, fileName string) (string, error) {
 	log.Debugf("Loaded %s", fileName)
 	return string(fileBytes), nil
 }
+
+// PreProcessPolicy fetches policy sources and returns a policy object with
+// pinned SHA/image digest URL where applicable, along with a policy cache object.
+func PreProcessPolicy(ctx context.Context, policyOptions policy.Options) (policy.Policy, *cache.PolicyCache, error) {
+	var policyCache *cache.PolicyCache
+	pinnedPolicyUrls := map[string][]string{}
+	policyCache, ok := PolicyCacheFromContext(ctx)
```

Comment:
This is a similar pattern to how we create _clients_ that make external connections. Consider introducing a helper function, `NewPolicyCache` that handles the logic of using an existing cache from the context, or creating a new one:  https://github.com/enterprise-contract/ec-cli/blob/0c2bc064788fbeacec82b7492e7330832a9b41d1/internal/kubernetes/client.go#L59-L74

Reply:
Done.